src/assets/pos.ts

```diff
@@ -76,39 +70,36 @@ export const formatInt = (num: number | bigint): string => {
 
 export const getValidatorKeys = async (
 	accounts: Account[],
-	snapshotHeight: number,
-	prevSnapshotBlockHeight: number,
 	db: Database,
 ): Promise<Record<string, string>> => {
-	const keys: Record<string, string> = {};
-
-	const blocksStream = db.createReadStream({
-		gte: Buffer.from(`${DB_KEY_BLOCKS_HEIGHT}:${formatInt(prevSnapshotBlockHeight + 1)}`),
-		lte: Buffer.from(`${DB_KEY_BLOCKS_HEIGHT}:${formatInt(snapshotHeight)}`),
-	});
-
-	const blockIDs: Buffer[] = await getBlocksIDsFromDBStream(blocksStream);
+	const delegateSet = new Set();
+	for (const account of accounts) {
+		if (account.dpos.delegate.username !== '') {
+			delegateSet.add(account.address.toString('hex'));
+		}
+	}
 
-	for (const id of blockIDs) {
-		const blockHeaderBuffer = await db.get(Buffer.from(`${DB_KEY_BLOCKS_ID}:${keyString(id)}`));
-		const blockHeader: BlockHeader = codec.decode(blockHeaderSchema, blockHeaderBuffer);
-		const payload = ((await getTransactions(id, db)) as unknown) as Transaction[];
+	const PAGE_SIZE = 100000;
```

Comment:
it would be flexible if could parametrise this? it allows us to easily change without having to make release if something needs changing dynamically

Reply:
Will address it as part of https://github.com/LiskHQ/lisk-migrator/pull/157.